### PR TITLE
Adding bitvector function definitions to the lean backend

### PR DESCRIPTION
**Pull request by lfrenot**
This adds a number of the bitvector operations to the lean backend.
The implemented operations are: eq, ne, append, add, sub, not, and, or, and xor.

The operations that are not implemented do not have a direct equivalent in the BitVec library and need wrappers.

### CODE REVIEW

**Comment by lfrenot** [comment]
This needs at least these two issues fixed before it can be merged:

- the sail `bool` type is printed in lean as `bool` instead of `Bool`
- the sail `truncate` function's arguments are `(bits, int)`,  but the lean `BitVec,truncate`'s arguments are `Nat, BitVec`

**Comment by tobiasgrosser** [comment]
Let's maybe first add the ones that work out-of-the-box. We can then fix the bugs 1:1.

**Comment by tobiasgrosser** [comment]
I would rather remove the implementation of the broken variants as well, rather than adding untested code, no?

**Comment by lfrenot** [comment]
Okay,
I wasn't sure which to do since the code is correct, the test is the only part that breaks

**Comment by lfrenot** [comment]
Oh, and I forgot to commit the edits to the test file anyways

**Comment by tobiasgrosser** [comment]
Still, if sth is not testable, I would not assume it to be correct. Its easier to add new code, than keep track of code that 'may still require test cases`.

**Comment by tobiasgrosser** [comment]
Can you just remove the code that cannot yet be tested. We do not want to commit commented-out code either, no?

File: test/lean/bitvec_operation.sail [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+val bitvector_or : (bits(16), bits(16)) -> bits(16)
+function bitvector_or(x, y) = {
+  x | y
+}
+
+val bitvector_xor : (bits(16), bits(16)) -> bits(16)
+function bitvector_xor(x, y) = {
+  xor_vec (x, y)
+}
```
Comment (tobiasgrosser) [inline test/lean/bitvec_operation.sail]: The file ending seems to be broken here, no?

Comment (lfrenot) [reply test/lean/bitvec_operation.sail]: I don't think so?

Comment (lfrenot) [reply test/lean/bitvec_operation.sail]: sail on my end does read the file correctly
But I guess the other test files do have extra endlines

Comment (tobiasgrosser) [reply test/lean/bitvec_operation.sail]: Github shows this red warning, no? So you probably want to add an additional newline at the end of the file.

**Pull request merged**